### PR TITLE
Panel operating mode defect

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**

### CODE REVIEW

File: src/bus_monitor.cpp [hunk 1]
```diff
@@ -670,129 +670,67 @@ void SystemStatus::listenBootProgressState()
...
+            }
         }
     }
 }
 
-void SystemStatus::listenSystemOperatingModeParameters()
+void SystemStatus::listenSystemOperatingMode()
 {
-    static auto sigPowerPolicy = std::make_unique<sdbusplus::bus::match::match>(
+    static auto biosMatcher = std::make_unique<sdbusplus::bus::match::match>(
```
Comment (Alpana07) [inline src/bus_monitor.cpp:713]: DO we need this local variable "biosMatcher"?

File: src/bus_monitor.cpp [hunk 2]
```diff
@@ -670,129 +670,67 @@ void SystemStatus::listenBootProgressState()
...
-            // std::cout << "power state = " << *powerState << std::endl;
-            powerPolicy = *powerState;
-
-            setSystemCurrentOperatingMode();
-        }
-        else
-        {
-            std::cerr << "Failed to read power policy from Dbus" << std::endl;
-        }
+        std::cerr << "Error in reading BIOS attribute signal " << std::endl;
```
Comment (Alpana07) [inline src/bus_monitor.cpp:677]: correct me if I m understanding it wrong.
IS this-  BIOS attribute value?
OR - BIOS attribute changed signal

File: include/bus_monitor.hpp [hunk 3]
```diff
@@ -273,19 +273,21 @@ class SystemStatus
...
-     * b) Power policy.
-     * c) Reboot policy.
+     * @brief BIOS attribute change callback.
+     *
+     * Listens to sytem operating mode property change in BIOS base table.
+     *
+     * @param[in] msg - Callback message.
      */
-    void setSystemCurrentOperatingMode();
+    void biosAttribsCallback(sdbusplus::message::message& msg);
```
Comment (Alpana07) [inline include/bus_monitor.hpp:290]: I have a suggestion in implementation file, please check

File: include/bus_monitor.hpp [hunk 4]
```diff
@@ -273,19 +273,21 @@ class SystemStatus
...
     /**
-     * @brief Set system operating mode.
-     * An api to set system operating mode based on the value of three
-     * parameters required to detect operating mode.
-     * a) Logging Setting.
-     * b) Power policy.
-     * c) Reboot policy.
+     * @brief BIOS attribute change callback.
+     *
+     * Listens to sytem operating mode property change in BIOS base table.
```
Comment (Alpana07) [inline include/bus_monitor.hpp:286]: can we mention here exact attribute name which we are listening  , like sytem_operating_mode property

**Review by Alpana07 — COMMENTED** [review]
I have few comments, please check

**Comment by Alpana07** [comment]
I have few comments, please check

File: src/bus_monitor.cpp [hunk 5]
```diff
@@ -670,129 +670,67 @@ void SystemStatus::listenBootProgressState()
         });
 }
 
-void SystemStatus::powerPolicyStateCallback(sdbusplus::message::message& msg)
+void SystemStatus::biosAttribsCallback(sdbusplus::message::message& msg)
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply src/bus_monitor.cpp]: Coding guidelines says not to keep name in short when the short name is not meaningful.
I can change to full form if it makes it more readable.

> also, by this name I am not understanding the functionality of it.
can we make it something like Init OR update BiosAttributes

This function is not initializing bios attributes neither it is updating bios attributes. It is a callback function which is triggered when there is a property change signal  received, accordingly it has been named.

File: include/bus_monitor.hpp (hunk 4 above)
Comment (SunnySrivastava1984) [reply include/bus_monitor.hpp]: Why? Internal names can be anything, the function is looking for 'a' property which denotes "system operating mode"

File: src/bus_monitor.cpp (hunk 2 above)
Comment (SunnySrivastava1984) [reply src/bus_monitor.cpp:677]: The callback is outcome of signal received.

File: src/bus_monitor.cpp (hunk 1 above)
Comment (SunnySrivastava1984) [reply src/bus_monitor.cpp:713]: It is not just a local variable.

File: include/bus_monitor.hpp (hunk 3 above)
Comment (SunnySrivastava1984) [reply include/bus_monitor.hpp]: Replied to that.

File: include/bus_monitor.hpp [hunk 6]
```diff
@@ -273,19 +273,21 @@ class SystemStatus
     void rebootPolicyStateCallback(sdbusplus::message::message& msg);
 
     /**
-     * @brief Api to initialize system operating parameters.
+     * @brief Api to initialize system operating mode.
+     *
+     * The API sets the state managers system state based on the operating mode
```
Comment (PriyangaRamasamy) [inline include/bus_monitor.hpp:278]: nit: state manager's

**Review by PriyangaRamasamy — COMMENTED** [review]
Have very minor comments. Code looks ok to me

**Comment by PriyangaRamasamy** [comment]
can you please paste the test section in commit message @SunnySrivastava1984

File: include/bus_monitor.hpp (hunk 6 above)
Comment (SunnySrivastava1984) [reply include/bus_monitor.hpp:278]: Done.

File: include/bus_monitor.hpp [hunk 7]
```diff
@@ -273,19 +273,21 @@ class SystemStatus
...
+     * at the tim eof panel coming up.
      */
-    void initSystemOperatingParameters();
+    void initSystemOpeartingMode();
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply include/bus_monitor.hpp:281]: Done

File: src/bus_monitor.cpp [hunk 8]
```diff
@@ -670,129 +670,67 @@ void SystemStatus::listenBootProgressState()
...
         {
-            // std::cout << "reboot state = " << *rebootState << std::endl;
-            rebootPolicy = *rebootState;
-
-            setSystemCurrentOperatingMode();
-        }
-        else
-        {
-            std::cerr << "Failed to read reboot policy from Dbus" << std::endl;
+            auto boisBaseTable = std::get<0>(property.second);
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply src/bus_monitor.cpp:688]: Done.

**Comment by SunnySrivastava1984** [comment]
> can you please paste the test section in commit message @SunnySrivastava1984

Done.

**Review by PriyangaRamasamy — APPROVED** [review]

**Review by Alpana07 — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**